### PR TITLE
Version Proof_type

### PR DESCRIPTION
**Pull request by psteckler**
Version `Transaction_snark.Proof_type`. Probably that won't change, but adhering to convention.

Also removed some `yojson`, `eq` annotations that weren't being used.

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [X] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/transaction_snark/transaction_snark.mli [hunk 1]
```diff
@@ -3,7 +3,15 @@ open Coda_base
 open Snark_params
 
 module Proof_type : sig
-  type t = [`Merge | `Base] [@@deriving bin_io, sexp, yojson]
+  module Stable : sig
+    module V1 : sig
+      type t = [`Base | `Merge] [@@deriving bin_io, sexp, yojson]
```
Comment (cmr) [inline src/lib/transaction_snark/transaction_snark.mli]: the order of the variants here is not the same. does the bin_io deriving care about the order in order to generate ids?

Comment (psteckler) [reply src/lib/transaction_snark/transaction_snark.mli]: I don't think it matters here. The section on polymorphic variants in https://github.com/janestreet/bin_prot, IIUC, that the tags are written as hashes of the tags themselves.

But I can change it, if you like, to eliminate that concern.

Comment (cmr) [reply src/lib/transaction_snark/transaction_snark.mli]: I think it's fine.

Comment (psteckler) [reply src/lib/transaction_snark/transaction_snark.mli]: Normalized all occurrences so that `Base` is first.

**Pull request merged**